Sources/BraintreeCore/Authorization/TokenizationKey.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+
+class TokenizationKey: ClientAuthorization {
+    
+    // MARK: - Internal Properties
+    
+    let type = AuthorizationType.tokenizationKey
+    let bearer: String
+    let configURL: URL
+    let originalValue: String
+    
+    // MARK: - Initializer
+    
+    init(_ rawValue: String) throws {
+        self.bearer = rawValue
+        self.originalValue = rawValue
+        guard let configURL = TokenizationKey.baseURLFromTokenizationKey(rawValue) else {
+            throw TokenizationKeyError.invalid
+        }
+        self.configURL = configURL
+    }
+    
+    // MARK: - Private Methods
+    
+    private static func baseURLFromTokenizationKey(_ tokenizationKey: String) -> URL? {
+        let pattern: String = "([a-zA-Z0-9]+)_[a-zA-Z0-9]+_([a-zA-Z0-9_]+)"
+        guard let regularExpression = try? NSRegularExpression(pattern: pattern) else { return nil }
+
+        let range = NSRange(location: 0, length: tokenizationKey.count)
+        let matches = regularExpression.matches(in: tokenizationKey, range: range)
+
+        if matches.count != 1 || matches.first?.numberOfRanges != 3 {
+            return nil
+        }
+
+        var environment: String = ""
+        var merchantID: String = ""
+
+        matches.forEach { match in
+            environment = (tokenizationKey as NSString).substring(with: match.range(at: 1))
+            merchantID = (tokenizationKey as NSString).substring(with: match.range(at: 2))
+        }
+
+        var components: URLComponents = URLComponents()
+        components.scheme = environment == "development" ? "http" : "https"
```

Comment:
This logic was just copy-pasted from BTAPIClient ([see original](https://github.com/braintree/braintree_ios/blob/da4c1056d4ae199917fb355f8c3ebe7118e3f03c/Sources/BraintreeCore/BTAPIClient.swift#L411-L413)). 

I don't know why we have this logic to use HTTP over HTTPS for the `development` env. It might be related to whatever localhost:3000 GW spin-up logic was used for this. I don't want to untangle it in this PR, but definitely a good thing to cleanup for a future PR (we should consider removing the `development` env altogether, I've never used it)

Reply:
If we want to make a follow up ticket to investigate/remove the development env that's fine with me. Not a blocker, just mentioned since other pieces of this code were touched, but can also be a follow up ticket.